### ISSUE TITLE
Inconsistent playlist state with repeat and consume

### ISSUE THREAD

**Issue by k00mi** [mopidy/mopidy#1512]
If both repeat and consume modes are enabled and the last remaining song in the playlist has been played (i.e. playlist is empty, no song is playing), mopidy still reports it is playing the last song.

mpc output will look something like this:
```
Daylight Dies - A Life Less Lived
[playing] #1/0   0:00/8:23 (0%)
volume: 66%   repeat: on    random: off   single: off   consume: on
```
Note the `[playing] #1/0`.

Connecting with ncmpcc once in this state leads to a failed assertion in ncmpcpp because it tries to highlight song 1 of 0. Connections that were already established before the end of the song are not affected.

Confirmed with mopidy versions 2.0.0 and 2.0.1

**Comment by adamcik** [comment]
Possibly an easier starter. Assumption is based on the idea that one could add a test for this based on one of the current ones and then relatively safely change the core behavior.

**Comment by sticken88** [comment]
I'm trying to reproduce this but I don't know how to enable repeat and consume modes. Can you please help me?

**Comment by k00mi** [comment]
@sticken88: When connected with ncmpcpp, press `r` for repeat and `R` for consume. Check the manual if you're using a different client.

**Comment by adamcik** [comment]
Also adding a unit test for this is a must for anyone fixing this.

**Comment by sticken88** [comment]
@adamcik can you please share hints on how to do that?

**Comment by alexjaw** [comment]
Have testet and could not reproduce the issue - ncmpcpp consumed the tracks and was also set in repeat mode. Was able to add new tracks without restarting the application. Is this issue for the latest builds? I have tested with mopidy 2.0.0.

**Comment by k00mi** [comment]
@alexjaw I updated the description to address your points.

**Comment by alexjaw** [comment]
Submitted a pull request.

**Comment by jodal** [comment]
Fixed by #1549.
